### PR TITLE
feat: edit data in nested table (PT-188211532)

### PR DESCRIPTION
**Pull request by emcelroy**
[#188211532](https://www.pivotaltracker.com/story/show/188211532)

### CODE REVIEW

**Review by tealefristoe — APPROVED** [review]
This looks good, but I do think it would be worth updating some types before merging.

I know you didn't add the `any`s that I pointed out, but this would be a good opportunity to fix them. I mentioned that you could just do something like `type Case = any`, but I bet something like this would work:
```
interface Case {
  id: number
  values: Values
}
```

I also think it would be better to use `Case`s instead of `CaseValuesWithId`.

I'm approving because I don't think this should hold up the PR if there's any urgency, but I do think it would be worth spending an hour or two to make these changes.

**Comment by emcelroy** [comment]
@tealefristoe Thanks 👍 I started improving the typing a bit. I'm pretty sure your other suggestions make sense, but I want to consult with @lublagg when she's back next week before implementing them. Since we're going to continue working in the same parts of the code, I'm going to merge this and we'll possibly make the other changes in the next PR.

**Pull request merged**
